src/components/AutoUpdateTime.js

```diff
@@ -0,0 +1,99 @@
+import {View} from 'react-native';
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import styles from '../styles/styles';
+import DateUtils from '../libs/DateUtils';
+import withLocalize, {withLocalizePropTypes} from './withLocalize';
+import Timers from '../libs/Timers';
+import Text from './Text';
+
+const propTypes = {
+    /** Timezone of the user from their personal details */
+    timezone: PropTypes.shape({
+        /** Value of selected timezone */
+        selected: PropTypes.string,
+
+        /** Whether timezone is automatically set */
+        automatic: PropTypes.bool,
+    }).isRequired,
+    ...withLocalizePropTypes,
+};
+
+class AutoUpdateTime extends PureComponent {
+    constructor(props) {
+        super(props);
+        this.getCurrentUserLocalTime = this.getCurrentUserLocalTime.bind(this);
+        this.updateCurrentTime = this.updateCurrentTime.bind(this);
+        this.getTimezoneName = this.getTimezoneName.bind(this);
+        this.state = {
+            timezone: this.getCurrentUserLocalTime(),
+        };
+    }
+
+    componentDidMount() {
+        this.updateCurrentTime();
+    }
+
+    componentDidUpdate() {
+        // Make sure the interval is up to date every time the component updates
+        this.updateCurrentTime();
+    }
+
+    componentWillUnmount() {
+        clearInterval(this.timer);
+    }
+
+    getCurrentUserLocalTime() {
+        return DateUtils.getLocalMomentFromDatetime(
+            this.props.preferredLocale,
+            null,
+            this.props.timezone.selected,
+        );
+    }
+
+    getTimezoneName() {
+        // With non-GMT timezone, moment.zoneAbbr() will return the name of that timezone, so we can use it directly.
+        if (Number.isNaN(Number(this.state.timezone.zoneAbbr()))) {
+            return this.state.timezone.zoneAbbr();
+        }
+
+        // With GMT timezone, moment.zoneAbbr() will return a number, so we need to display it as GMT {abbreviations} format, ie: GMT +07
+        return `GMT ${this.state.timezone.zoneAbbr()}`;
+    }
+
+    /**
+     * Update the user's local time at the top of every minute
+     */
+    updateCurrentTime() {
+        if (this.timer) {
+            clearInterval(this.timer);
+            this.timer = null;
+        }
+        const secondsUntilNextMinute = (60 - this.state.timezone.seconds()) * 1000;
+        this.timer = Timers.register(
+            setInterval(() => {
```

Comment:
Since we are setting the clearing the interval each time and setting it again to the "seconds left for the next minute", shouldn't we be using `setTimeout` instead? the callback ends up being called just once.

Reply:
I remembered that we avoid to use setTimeout right, that's one of the reason I didn't use the setTimeout here.

Reply:
hmm, but using `setInterval` is like the same here, is there an advantage of `setInterval` over `setTimeout`. I image. that the "avoid setTimeout` would apply to `setInterval` too :P, but maybe I'm missing something 🤔

Reply:
Yeah, it's quite similar since the Interval only called once, since we avoid using setTimeout, I think we can use setInterval for now.

Reply:
For the given implementation it makes sense to use `setTimeout`. Here we're unnecessarily calling `clearTimeout`, when the setTimeout could've sufficed.

and I agree with @aldo-expensify, when we say avoid `setTimeout`, we mean avoid calls kinds of workarounds to get async code working, which also includes `setInterval`

Reply:
Thanks guys, I updated my PR @mananjadhav @aldo-expensify